### PR TITLE
v3.0.0

### LINKED ISSUE #91: Can I use wavesurfer with videojs-playlist?
Is it possible to use wavesurfer with the [videoplaylist](https://github.com/brightcove/videojs-playlist) plugin? I can't figure out how to merge the two.

Is it possible to attach wavesurfer like this and update the waveform depending on what's playing in the playlist?

```
vidplayer.wavesurfer({
	src: 'can this be fetched from the playlist?',
	msDisplayMax: 10,
	debug: true,
	waveColor: 'grey',
	progressColor: 'black',
	cursorColor: 'black',
	hideScrollbar: true
});
```

I'm getting this error when I'm trying the above approach:
> VIDEOJS: ERROR: TypeError: "this.player.options_.plugins is undefined"

### LINKED ISSUE #93: Video.js play/pause toggle broken with wavesurfer MediaElement source
## Description
When using Wavesurfer's MediaElement source via videojs-wavesurfer the play/pause toggle in videojs breaks. The native media controls still work fine, calling the video.js player object's `play()` & `pause()` functions in the console also works.

## Steps to reproduce
Open this jsfiddle:

https://jsfiddle.net/crgdm3vx/

Hit the play icon, nothing happens.

In case the above jsfiddle goes offline:

```html
<!doctype html>
<html>
<head>
  <meta charset="utf-8">
  <title>Wavesurfer Plugin Video Example</title>

  <link href="https://vjs.zencdn.net/7.6.5/video-js.min.css" rel="stylesheet">
  <link href="https://collab-project.github.io/videojs-wavesurfer/dist/css/videojs.wavesurfer.min.css" rel="stylesheet">

  <script src="https://vjs.zencdn.net/7.6.5/video.min.js"></script>
  <script src="https://unpkg.com/wavesurfer.js"></script>

  <script src="https://collab-project.github.io/videojs-wavesurfer/dist/videojs.wavesurfer.min.js"></script>

  <style>
  /* change player background color */
  #myVideo {
      background-color: #6699FF;
  }
  </style>
</head>
<body>

<video id="myVideo" playsinline class="video-js vjs-default-skin">
  <source src="https://collab-project.github.io/videojs-wavesurfer/examples/media/example.mp4"></source>
</video>

<div id="waveform"></div>

<script>

videoElt = document.querySelector('#myVideo');

var player;
var options = {
    controls: true,
    autoplay: false,
    loop: false,
    width: 600,
    height: 300,
    plugins: {
        wavesurfer: {
            src: videoElt,
            container: '#waveform',
            msDisplayMax: 10,
            debug: true,
            waveColor: '#336699',
            progressColor: 'black',
            cursorColor: 'black',
            hideScrollbar: true,
            // disable options below to hide the video playback,
            // and only show the waveform
            backend: 'MediaElement',
        }
    }
};
  // create player
  player = videojs('myVideo', options, function() {
    // print version information at startup
    var msg = 'Using video.js ' + videojs.VERSION +
        ' with videojs-wavesurfer ' +
        videojs.getPluginVersion('wavesurfer') +
        ' and wavesurfer.js ' + WaveSurfer.VERSION;
    videojs.log(msg);
  });

  player.on('waveReady', function(event) {
    console.log('waveform: ready!');
  });

  player.on('playbackFinish', function(event) {
    console.log('playback finished.');
  });

  // error handling
  player.on('error', function(error) {
    console.warn('ERROR:', error);
  });
</script>

</body>
</html>
```

### Expected
The video should play/pause according to the state of the MediaElement.

### Actual
Nothing.

If debug logging is enabled the "VIDEOJS: Pause playback" & "VIDEOJS: Start playback" messages are written to the javascript console.

### Error output
N.A.

## Versions
Make sure to include the following versions:
### videojs/wavesurfer
video.js: 7.6.5
videojs.wavesurfer: 2.11.0
wavesurfer.js: 3.1.0
### Browsers
Firefox, Chromium
### OSes
Fedora, Arch Linux

### PR DESCRIPTION
**Pull request by thijstriemstra**
This requires wavesurfer.js 3.2.0 or newer.

fixes #93 
fixes #91

### CODE REVIEW

**Comment by ybinnenwegin2ip** [comment]
This PR solves issue #93 

With a few adjustments to my example as noted in the v3 branch's examples directory both audio and video playback seem to work fine via MediaElement. I am also able to put the video's waveform in a separate `div` using the `container` option for wavesurfer.

Thanks a ton @thijstriemstra !

File: test/videojs.wavesurfer.spec.js [hunk 1]
```diff
@@ -275,78 +296,64 @@ describe('Wavesurfer', () => {
...
-                TestHelpers.EXAMPLE_AUDIO_PEAKS_FILE
-            );
-        });
+        player.one(Event.WAVE_READY, done);
+
+        player.src({
+            src: TestHelpers.EXAMPLE_AUDIO_FILE,
+            type: TestHelpers.EXAMPLE_AUDIO_TYPE,
+            peaks: TestHelpers.EXAMPLE_AUDIO_PEAKS_FILE
+        })
```
Comment (EricTendian) [inline test/videojs.wavesurfer.spec.js]: I was trying to build this locally, but it looks like this is missing a semicolon, so the linter fails. Can this be fixed (no rush)? 🙂

Comment (thijstriemstra) [reply test/videojs.wavesurfer.spec.js]: you can always make a pull request to fix it ;)

I fixed it for you.

**Comment by thijstriemstra** [comment]
Support for `MediaElement` backend is fixed now. WebAudio backend (i.e. the original implementation of videojs-wavesurfer) is next.

**Comment by thijstriemstra** [comment]
> . I am also able to put the video's waveform in a separate div using the container option for wavesurfer.

if you can improve the video example in this branch like that, help is always welcome @ybinnenwegin2ip

**Comment by thijstriemstra** [comment]
TODO: how to distribute/enable the middleware in the plugin, see https://github.com/videojs/video.js/issues/6311

**Pull request merged**